code/datums/elements/tool_flash.dm

```diff
@@ -33,5 +33,5 @@
 /datum/element/tool_flash/proc/flash(datum/source, mob/living/user)
 	SIGNAL_HANDLER
 
-	if(user && get_dist(get_turf(source), get_turf(user)) <= 1)
+	if(user && get_dist(get_turf(source), get_turf(user)) <= 1 && flash_strength > 0)
```

Comment:
```suggestion
	if(user && flash_strength > 0 && get_dist(get_turf(source), get_turf(user)) <= 1)
```

better shortcircuit to avoid calling get_dist unless necessary (literally irrelevant microoptimization but it reads better to do valid user, then a simple integer check, then a proc call to my eyes so double bonus)